src/request-queue.js

```diff
@@ -53,24 +61,26 @@ export class RequestQueue {
             if (this.isPolling) {
                 this.poll()
             }
-        }, POLL_INTERVAL)
+        }, this._pollInterval)
     }
 
     unload() {
-        clearInterval(this._poller)
+        clearTimeout(this._poller)
         const requests = this._event_queue.length > 0 ? this.formatQueue() : {}
         this._event_queue.length = 0
         for (let url in requests) {
-            this.handlePollRequest(url, requests[url], { unload: true })
+            const { data, options } = requests[url]
+            this.handlePollRequest(url, data, { ...options, transport: 'sendbeacon' })
         }
     }
 
     formatQueue() {
         const requests = {}
         _.each(this._event_queue, (request) => {
-            const { url, data } = request
-            if (requests[url] === undefined) requests[url] = []
-            requests[url].push(data)
+            const { url, data, options } = request
+            const key = options?._batchKey || url
```

Comment:
This line gave me some problems:

![image](https://user-images.githubusercontent.com/53387/100153806-f2536d00-2ea4-11eb-8270-d40b79f67cde.png)

What's the chance `options` is not an object? When I tried, 3 tests failed and there seemed to be a rabbit hole if I defaulted it to `{}` in `enqueue`.

There's another rabbit hole to go down in if you'd like to enable babel to transpile this into slightly older JS... though I'd just bail and `? :` here..

Reply:
Interesting.

`options` definetly may not be an object, in this case jest tests actually tested this and didn't catch an error.

Updated it, thanks!